symengine/printer.h

```diff
@@ -33,8 +33,12 @@ class Precedence : public BaseVisitor<Precedence>
     template <typename Poly>
     void bvisit_upoly(const Poly &x)
     {
-        if (x.get_dict().size() == 1) {
-            auto it = x.get_dict().begin();
+        auto begin = x.begin();
+        auto it = x.begin();
+        auto end = x.end();
+
+        if (end == ++begin) {
```

Comment:
The "polynomial size" is defined as `0` if it's a zero poly otherwise `degree+1`  (as used by flint)
This was the only way I can think of to make it work for all the upolys.